ext/nnstreamer/tensor_filter/tensor_filter_edgetpu.cc

```diff
@@ -0,0 +1,400 @@
+  if (tflite::InterpreterBuilder(model, resolver)(&interpreter) !=
+      kTfLiteOk) {
+    std::cerr << "Failed to build interpreter." << std::endl;
+  }
+
+  interpreter->SetExternalContext(kTfLiteEdgeTpuContext, edgetpu_context);
+  interpreter->SetNumThreads(1);
+  if (interpreter->AllocateTensors() != kTfLiteOk) {
+    std::cerr << "Failed to allocate tensors." << std::endl;
+  }
+
+  return interpreter;
+}
+
+static void
+edgetpu_close (const GstTensorFilterProperties * prop, void **private_data);
+
+/**
+ * @brief Configure private_data
+ */
+static int
+allocateData (const GstTensorFilterProperties * prop, void **private_data)
+{
+  pdata *data;
+  if (*private_data != NULL) {
+    /* Already opened */
+    data = (pdata *) *private_data;
+
+    if (!prop->model_files[0] || prop->model_files[0][0] == '\0') {
+      std::cerr << "Model path is not given." << std::endl;
+      return -EINVAL;
+    }
+    if (data->model_path && g_strcmp0 (prop->model_files[0],
+            data->model_path) == 0) {
+      return 0; /* Already opened with same file. Skip ops */
+    }
+
+    edgetpu_close (prop, private_data); /* Close before opening one. */
+  }
+
+  *private_data = data = g_new0 (pdata, 1);
+  if (data == NULL) {
+    std::cerr << "Failed to allocate memory for edge-tpu tensor_filer."
+        << std::endl;
+    return -ENOMEM;
+  }
+  return 0;
+}
+
+/**
+ * @brief from tflite-core
+ * @todo Remove this func or make them shared
+ */
+static int
+getTensorDim (tflite::Interpreter *interpreter, int tensor_idx,
+    tensor_dim dim)
+{
+  TfLiteIntArray *tensor_dims = interpreter->tensor (tensor_idx)->dims;
+  int len = tensor_dims->size;
+  g_assert (len <= NNS_TENSOR_RANK_LIMIT);
+
+  /* the order of dimension is reversed at CAPS negotiation */
+  std::reverse_copy (tensor_dims->data, tensor_dims->data + len, dim);
+
+  /* fill the remnants with 1 */
+  for (int i = len; i < NNS_TENSOR_RANK_LIMIT; ++i) {
+    dim[i] = 1;
+  }
+
+  return 0;
+}
+
+/**
+ * @brief From tflite-core.cc
+ * @todo Remove this or make them shared
+ */
+static tensor_type
+getTensorType (TfLiteType tfType)
+{
+  switch (tfType) {
+    case kTfLiteFloat32:
+      return _NNS_FLOAT32;
+    case kTfLiteUInt8:
+      return _NNS_UINT8;
+    case kTfLiteInt32:
+      return _NNS_INT32;
+    case kTfLiteBool:
+      return _NNS_INT8;
+    case kTfLiteInt64:
+      return _NNS_INT64;
+    case kTfLiteString:
+    default:
+      /** @todo Support other types */
+      break;
+  }
+
+  return _NNS_END;
+}
+
+/**
+ * @brief extract and store the information of given tensor list
+ * @param tensor_idx_list list of index of tensors in tflite interpreter
+ * @param[out] tensorMeta tensors to set the info into
+ * @return 0 if OK. non-zero if error.
+ */
+int
+setTensorProp (tflite::Interpreter *interpreter,
+    const std::vector<int> &tensor_idx_list,
+    GstTensorsInfo * tensorMeta)
+{
+  tensorMeta->num_tensors = tensor_idx_list.size ();
```

Comment:
add a check that num_tensors less than NNS_TENSOR_SIZE_LIMIT

Reply:
Fixed.